### PR TITLE
add proj centroid

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):**

- n/a

**Proposed Changes:**

1. Add Projection Extension field proj:centroid

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/stactools/naip/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

**Review by gadomski — DISMISSED** [review]

**Comment by philvarner** [comment]
These tests pass locally, but fail in CI, so I'm going to have to figure out what the difference is.

**Comment by gadomski** [comment]
> These tests pass locally, but fail in CI, so I'm going to have to figure out what the difference is.

I'd look at shapely version first?

**Comment by philvarner** [comment]
I don't think is shapely -- for whatever reason, the calculation of the centroid returns an empty point, and I can't repro it locally, even after creating that same conda env 🤷🏻

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -51,8 +51,11 @@ jobs:
...
         with:
           auto-update-conda: true
+          environment-file: environment.yml
           python-version: ${{ matrix.python-version }}
-      - name: Update Conda's environemnt
-        run: conda env update -f environment.yml -n test
+      - name: update certs so PROJ can download the us_noaa_cshpgn.tif file
+        run: |
+          sudo mkdir -p /etc/pki/tls/certs
+          sudo curl https://curl.se/ca/cacert.pem -o /etc/pki/tls/certs/ca-bundle.crt
```
Comment (gadomski) [inline .github/workflows/continuous-integration.yml:59]: This is wild, I can't believe it hasn't come up before. But so it goes, good catch.

**Review by gadomski — APPROVED** [review]

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (philvarner) [reply .github/workflows/continuous-integration.yml:59]: I think likely it's specific to NAD83 / UTM zones that NAIP use -- I don't know of any of the other packages that use that projection. I think the test also never test if the reprojected geometry is correct, so the (Inf, Inf) point was never asserted, and it didn't actually cause a test failure til I tried to use the centroid.

Comment (gadomski) [reply .github/workflows/continuous-integration.yml:59]: I was more referring to the "need to download certs" issue.

Comment (philvarner) [reply .github/workflows/continuous-integration.yml:59]: that's what I mean -- I think NAD83 / UTM zones are the only ones that require PROJ to actually download something over HTTPS, so none of the other stactools packages need the certs.

Comment (gadomski) [reply .github/workflows/continuous-integration.yml:59]: Gotcha -- I guess maybe I'm surprised Github's runners aren't set up to download SSL out-of-the-box? Anyways, all good.

**Pull request merged**